Robot_Adapter/Convert/FromRobot/Loads/BarVaryingDistributedLoad.cs

```diff
@@ -0,0 +1,99 @@
+            double fay = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_PY1);
+            double faz = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_PZ1);
+            double distA = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_X1);
+            double fbx = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_PX2);
+            double fby = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_PY2);
+            double fbz = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_PZ2);
+            double distB = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_X2);
+
+            double rel = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_RELATIVE);
+            double axis = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_LOCAL);
+            double proj = loadRecord.GetValue((short)IRobotBarTrapezoidaleRecordValues.I_BTRV_PROJECTION);
+
+            if (rel != 0)
+            {
+                Engine.Reflection.Compute.RecordWarning("Currently no support for BarVaryingDistributedLoad with relative distance from ends");
+                return null;
+            }
+
+            return new BarVaryingDistributedLoad
+            {
+                ForceA = new Vector { X = fax, Y = fay, Z = faz },
+                ForceB = new Vector { X = fbx, Y = fby, Z = fbz },
+                DistanceFromA = distA,
+                DistanceFromB = distB,
+                Axis = axis.FromRobotLoadAxis(),
+                Projected = proj.FromRobotProjected()
+            };
+        }
+
+        /***************************************************/
+
+        //Fixing the issue with Robot defining the second point in the BarVaryingLoad in relation to the start, while BHoM defines it in relation to the end
+        public static List<BarVaryingDistributedLoad> FixVaryingLoadEndDistances(this BarVaryingDistributedLoad load)
+        {
+            List<BarVaryingDistributedLoad> loads = new List<BarVaryingDistributedLoad>();
+
+            int counter = 0;
+            foreach (var lengthGroup in load.Objects.Elements.GroupBarsByLength(0.001))
+            {
+                BarVaryingDistributedLoad clone = load.GetShallowClone() as BarVaryingDistributedLoad;
+                clone.DistanceFromB = lengthGroup.Key - clone.DistanceFromB; //Set distance from B to be from end node instead of start
+                clone.Objects = new BHoMGroup<Bar>() { Elements = lengthGroup.Value };
+                loads.Add(clone);
+                counter++;
+            }
+
+            if (counter > 1)
+                Engine.Reflection.Compute.RecordNote("Varying BarLoads in BHoM meassures distans from start for the first point and from end for the second point, whilst Robot measures only from start node. To accomodate this, load pulled from Robot as been split up in multiple loads.");
```

Comment:
```suggestion
                Engine.Reflection.Compute.RecordNote("Varying BarLoads in BHoM measures distance from start for the first point and from end for the second point, whilst Robot measures only from start node. To accommodate this, load pulled from Robot as been split up in multiple loads.");
```

Reply:
I also don't get from the message nether why or how that difference lead to multiple loads (or how those multiple loads would look)? Can't we measure the Bar's length?

Checked the code after being confused and see that you mean that one load no longer encompasses all the bars it used to. Some clarification might be good